### ISSUE TITLE
🐛 BUG:

### ISSUE THREAD

**Issue by mavilein** [cloudflare/workers-sdk#5623]
### Which Cloudflare product(s) does this pertain to?

Wrangler core, Miniflare

### What version(s) of the tool(s) are you using?

3.46.0

### What version of Node are you using?

21.7.3

### What operating system and version are you using?

Mac Ventura 13.4.1

### Observed behavior
We noticed a regression in accessing a Durable Object in another Worker Project when runnig locally after upgrading to 3.46.0. After this upgrade all calls to the Durable Object are failing with opaque errors. All responses result in a 500 without useful response bodies or headers.
We are only observing this behavior locally and not in production. This only occurs in stub calls for DOs that are part of a different worker project.

### Expected behavior
Calls to a DO should still succeed even after upgrading to a recent wrangler version.

### Steps to reproduce
Please check the repository with the attached reproduction. We confirmed via testing that this issue only starts to appear with `3.46.0`. Here is a super quick gist of the issue. All calls to `stub.fetch` with the following shape fail:
```ts
const response = await stub.fetch(request.url, {
	method: request.method,
	headers: request.headers,
	body: request.body,
});
```

We were able to unblock ourselves by copying over the `cf` property. The absence of this property leads to crashes in workerd/miniflare.

```ts
const response = await stub.fetch(request.url, {
	method: request.method,
	headers: request.headers,
	body: request.body,
	// if the following gets copied over it works again
	cf: request.cf
});
```

### Please provide a link to a minimal reproduction

https://github.com/prisma/reproduction-cf-do-fetch-broken

**Comment by bruceharrison1984** [comment]
I am seeing similar behavior from `wrangler 3.50.0`. Web requests fail with no discernible error, other than the request has failed.

```ts
// typical miniflare request
  return wranglerDevServer.fetch(url, {
    headers,
    method,
  });
```

When viewing requests from the browser, the only response recorded is `net::ERR_CONTENT_DECODING_FAILED`.  Rolling back to `3.49.0` immediately resolves the issue.

**Comment by CarmenPopoviciu** [comment]
hi folks, 

this might have been fixed by https://github.com/cloudflare/workers-sdk/pull/5798. Can you please update your wrangler version and let us know if you're still experiencing issues 🙏 ?

**Comment by petebacondarwin** [comment]
Closing due to no response. We assume that this has been fixed by #5798. Please comment here or open a new issue if that is not the case.
